File: 3248. Snake in Matrix/Solution.py
from typing import List


class Solution:
    def finalPositionOfSnake(self, n: int, commands: List[str]) -> int:
        i = j = 0
        for s in commands:
            match s[0]:
                case "U":
                    i -= 1
                case "D":
                    i += 1
                case "L":
                    j -= 1
                case "R":
                    j += 1
        return i * n + j

File: 3248. Snake in Matrix/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int finalPositionOfSnake(int n, vector<string>& commands)
    {
        int i = 0, j = 0;
        for (const string& s : commands) {
            switch (s[0]) {
            case 'U':
                --i;
                break;
            case 'D':
                ++i;
                break;
            case 'L':
                --j;
                break;
            default:
                ++j;
            }
        }
        return i * n + j;
    }
};
